File: A/5th day/Juicer/Juicer (Eman Elsayed).dart
import 'dart:io';

main() {
  // take input
  int? n = int.parse(stdin.readLineSync()!);
  int? b = int.parse(stdin.readLineSync()!);
  int? d = int.parse(stdin.readLineSync()!);
  int sum = 0;
  int count = 0;
  for (int i = 0; i < n; i++) {
    int? x = int.parse(stdin.readLineSync()!); // n items
    if (x <= b) {
      // if the size of the orange is less than or equal to the size of the juicer
      sum += x; // add the size of the orange to the sum
    }
    if (sum > d) {
      // if the sum of the oranges is greater than the size of the waste
      sum = 0;
      count++; // empty the waste
    }
  }
  print(count); // print the number of times the waste was emptied
}

File: A/5th day/Juicer/Juicer (Eman Elsayed).cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n, b, d; // n = number of oranges , b = max size of orange Juicer , d = max size of waste box
    cin >> n >> b >> d;
    vector<int> vec(n);
    for (auto &i : vec)
        cin >> i;
    int sum = 0, cnt = 0;
    for (int i = 0; i < n; i++)
    {
        if (vec[i] <= b) // if the orange size is less than or equal to the max size of orange Juicer it will be juiced
        {
            sum += vec[i]; // sum the size of the orange to the waste box
            if (sum > d)   // if the waste box is full it will be emptied
            {
                sum = 0;
                cnt++; // count the number of times the waste box is emptied
            }
        }
    }
    cout << cnt << endl; // print answer (number of times the waste box is emptied)
    return 0;
}